sensors/omron_d6t/omron_d6t.py

```diff
@@ -0,0 +1,96 @@
+import time, crcmod.predefined
+from sensors.sensor import Sensor
+
+class OmronD6T(Sensor):
+    __START_COMMAND = 0x4c
+    __ARRAY_SIZE = 16
+    __BUFFER_LENGTH = 35
+    __PEC_BUFFER = [0x15]
+    __MAX_CELLS_ABOVE_ROOM_TEMPERATURE = 10
+
+    def __init__(self, bus = 1, address = 0x0a):
+        super().__init__(bus, address, self.__START_COMMAND)
+
+    @property
+    def buffer_length(self):
+        return self.__BUFFER_LENGTH
+
+    @property
+    def occupied_status(self):
+        room_temperature, temperature_matrix = self.__read()
+
+        return self.__temperature_cells_above_threshold(room_temperature, temperature_matrix) >= self.__MAX_CELLS_ABOVE_ROOM_TEMPERATURE
+
+    def __temperature_cells_above_threshold(self, room_temperature, temperature_array):
+        """
+        :param room_temperature: Room temperature
+        :param temperature_array: Temperature list from sensor
+
+        :return: Number of cells above room temperature 
+        """
+        return sum(temperature > room_temperature for temperature in temperature_array)
+
+    def __read(self):
+        """
+        Read sensor data and parse it appropriately. Checks PEC to ensure the data is valid.
+
+        :return: Tuple containing room temperature and an array of measured temperatures, or None if there was an issue
+        reading from the sensor
+        """
+
+        # This loop will run until a valid value is received from the sensor (or max retries is reached)
+        for i in range(0, self.MAX_RETRIES):
+            bytes_read, sensor_raw_data = self.read_raw_data()
+
+            if bytes_read == self.buffer_length:
```

Comment:
To me this would make more sense if it was clear that `buffer_length` was a constant. Now I'm left questioning if it's a dynamic value.

Reply:
Good point, I'll update. I initially added the method so that the superclass could call it when reading raw data and made more sense to "enforce" the implementation of a method rather than enforcing the definition of a constant

Reply:
Maybe it can be passed into the super-constructor during construction. I agree that asking sub-classes to overwrite constants is strange.

Reply:
^ That's what I was thinking as well Dom